### PR TITLE
Add ability to use Figleaf::Config.define do....

### PR DESCRIPTION
**Pull request by jcmuller**
You can express your configuration in ruby code:

```ruby
Figleaf::Config.define do
  root_key "value"

  default do
    everywhere "value"
  end

  test do
    foo "bar"
  end
end
```

### CODE REVIEW

**Comment by jcmuller** [comment]
Solves #12

**Comment by delwaterman** [comment]
GTG

File: lib/figleaf/config.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  class Config
+    def initialize
+      @property = {}
+    end
+
+    def self.define(property_name, &block)
+      property = new.define(&block)
+
+      Settings.configure_with_auto_define do
+        if Settings.respond_to?(property_name) &&
```
Comment (jcmuller) [inline lib/figleaf/config.rb]: Infer name from file name if file location is `config/settings`. Otherwise, require property name to be passed in.

**Comment by agibralter** [comment]
❤️

**Pull request merged**